design/gateway-apis-implementation.md

```diff
@@ -1,204 +1,420 @@
+The most important part here is that the field is only used if the AllowRouteOverride field is set in the referencing Gateway resource.
 
-Errors or conflicts here will render that rule invalid, but not the rest of the rules.
-Other valid rules will still be passed to Envoy.
-For each invalid rule, Contour will update status information with the rule and the reason.
-A conflict will **not** result in the whole HTTPRoute being rejected unless there are zero rules left.
+Contour will only ever update the status of HTTPRoute objects.
 
+Errors or conflicts here will render that rule invalid, but not the rest of the rules. Other valid rules will still be passed to Envoy.
+For each invalid rule, Contour will update status information with the rule and the reason. A conflict will not result in the whole HTTPRoute being rejected unless there are zero rules left.
 
 The output of this watcher is Envoy configuration.
 
 ### TLSRoute
-Contour will watch all TLSRoute objects, and filter for entries matching a label selector in its configured Gateway's `spec.listeners[].routes`,
-and will configure the associated routes.
-Configuration of TLSRoutes  will be subject to the rules around the `RouteGateways` field.
 
-Contour will only ever update the `status` of TLSRoute objects.
+Contour will watch all TLSRoute objects, and filter for entries matching a label selector in its configured Gateway's spec.listeners[].routes, and will configure the associated routes.
+Configuration of TLSRoutes will be subject to the rules around the RouteGateways field.
 
-Errors or conflicts here will render that section of the config invalid.
-Other valid sections will still be passed to Envoy.
+Contour will only ever update the status of TLSRoute objects.
+
+Errors or conflicts here will render that section of the config invalid. Other valid sections will still be passed to Envoy.
 
 The output of this watcher is Envoy configuration.
 
 ## Detailed Design
 
-### Configuration
-Contour will have an entry added to the config file for the Gateway it should watch.
+### Responsibilities
+
+#### Contour
+
+In the fully managed mode, Contour is responsible for managing the Envoy fleet which includes the Envoy Kubernetes Daemonset and the Envoy Kubernetes Service.
+
+#### Operator
+
+The operator is responsible for managing instances of Contour as well as any additional configuration items that are related to running Contour or Envoy.
+These pieces will be configured in the Contour Configuration CRD.
+
+### Configmap to CRD
+
+Currently, Contour gets its configuration from two different places, one is the configuration file represented as a Kubernetes configmap.
+The other are flags which are passed to Contour.
+
+This design will migrate the Contour configuration configmap to a full custom resource definition which will allow for better validation, easier processing, as well as represent a full Contour configuration including how Envoy is to be provisioned if the user desires.
```

Comment:
Are we planning to continue supporting the config file as well until Contour 2.0? Otherwise seems like this would constitute a breaking change.

Reply:
I think this would only support the crd. So we'd deprecate support for managed bits using a configmap.

Reply:
I agree, I think the stance here is:
- Contour proper continues supporting the config file until we release a 2.0 (which may be after October), but it's deprecated.
- The fully-managed, operator flow requires the CRD before then.